zebra/rt_netlink.c

```diff
@@ -246,6 +246,42 @@ static vrf_id_t vrf_lookup_by_table(uint32_t table_id, ns_id_t ns_id)
 	return VRF_DEFAULT;
 }
 
+/**
+ * @parse_encap_mpls() - Parses encapsulated mpls attributes
+ * @tb:         Pointer to rtattr to look for nested items in.
+ * @labels:     Pointer to store labels in.
+ *
+ * Return:      Number of mpls labels found.
+ */
+static int parse_encap_mpls(struct rtattr *tb, mpls_label_t *labels)
+{
+	struct rtattr *tb_encap[MPLS_IPTUNNEL_MAX + 1] = {0};
+	mpls_lse_t *lses = NULL;
+	int num_labels = 0;
+	uint32_t ttl = 0;
+	uint32_t bos = 0;
+	uint32_t exp = 0;
+	mpls_label_t label = 0;
+
+	netlink_parse_rtattr_nested(tb_encap, MPLS_IPTUNNEL_MAX, tb);
+	lses = (mpls_lse_t *)RTA_DATA(tb_encap[MPLS_IPTUNNEL_DST]);
+	while (!bos && num_labels < MPLS_MAX_LABELS) {
+		mpls_lse_decode(lses[num_labels], &label, &ttl, &exp, &bos);
+		/* Encapped mpls messages come with a ttl associated with them
+		 * If there isn't one alread present in the label itself, we
+		 * re-encode it here with the one from the encap command.
+		 */
+		if (!ttl && tb_encap[MPLS_IPTUNNEL_TTL]) {
+			ttl = *(uint32_t *)RTA_DATA(
+				tb_encap[MPLS_IPTUNNEL_TTL]);
+			mpls_lse_decode(mpls_lse_encode(label, ttl, exp, bos),
+					&label, &ttl, &exp, &bos);
```

Comment:
Thanks for the updates. I only don't understand this part of the diff. Can't we just ignore `MPLS_IPTUNNEL_TTL` at all since we don't care about the ttl?

Reply:
yep removing.